pkg/daemon/daemon.go

```diff
@@ -500,35 +499,25 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 			glog.Errorf("nodeStateSyncHandler(): plugin %s error: %v", k, err)
 			return err
 		}
-
-		if p.Name() == "mco_plugin" && d && !r {
-			mcoReboot = true
-		}
-
 		glog.V(0).Infof("nodeStateSyncHandler(): plugin %s: reqDrain %v, reqReboot %v", k, d, r)
 		reqDrain = reqDrain || d
 		reqReboot = reqReboot || r
 	}
 	glog.V(0).Infof("nodeStateSyncHandler(): reqDrain %v, reqReboot %v disableDrain %v", reqDrain, reqReboot, dn.disableDrain)
 
 	for k, p := range dn.LoadedPlugins {
-		if k != GenericPlugin && k != McoPlugin {
+		if k != GenericPlugin {
+			if k == K8sPlugin && utils.ClusterType == utils.ClusterTypeOpenshift {
```

Comment:
Is there a reason why this condition needed for the k8s plugin and openshift cluster type?
 on line 663 you ensure that plugin is not loaded

Reply:
Will remove, there is no need for this conditional check.

Reply:
Removed.